src/main/java/tennisclub/entity/Court.java

```diff
@@ -40,20 +46,25 @@ public void setAddress(String location) {
         return Collections.unmodifiableSet(events);
     }
 
-    public void addEvent(Event event) {
-        events.add(event);
-        event.setCourt(this);
+    public String getPreviewImage() {
+        return previewImage;
+    }
+
+    public void setPreviewImage(String previewImage) {
+        this.previewImage = previewImage;
     }
 
     @Override
     public boolean equals(Object obj) {
-        // TODO implement properly
-        return obj instanceof Court && Objects.equals(id, ((Court) obj).id);
+        if (!(obj instanceof Court)) {
+            return false;
+        }
+        Court court = (Court) obj;
+        return this == court || id != null && id.equals(court.id);
     }
 
     @Override
     public int hashCode() {
-        // TODO implement properly
         return Objects.hash(id);
```

Comment:
Auto-generated ID should not be a part of equals and hashCode

Reply:
The problem is that Court has no natural candidate key. Should the methods be implemented simply like this?
```java
public boolean equals(Object obj) { return false; }
public int hashCode() { return SOME_INTEGER_CONSTANT; }
```

Or should we add some unique non-generated attribute, like a human-readable court name?

Reply:
(I assume there can be multiple courts with the same address, e.g. in one sports center.)

Reply:
Hmm... In this case I believe we should not use auto generated value and let the user to manage the court number (and use your original implementation of these methods)

Reply:
In this case I think it might be better to create an another attribute _(for example, court number)_, however, to prevent unnecessary duplication of location due to the possibility of having more courts on the same address, we would have to separate the location into a separate entity so we would reduce the redundancy of location. What do you think? 🤔